resources/main.js

```diff
@@ -48,6 +49,23 @@ function copyToClipboard(temptext) {
         tempItem.focus();
         tempItem.select();
         document.execCommand('copy');
-        mw.notify( mw.message( 'scratch-confirmaccount-click-copy-alert', { autoHide: true }, {autoHideSeconds: 5}) ); // Use an i18n message to send a notification
         tempItem.parentElement.removeChild(tempItem);
       }
+
+
+$(function () {
+    const elem = document.getElementsByName("scratchusername")[0];
+    if (!elem) return;
+
+    elem.onblur = function() {
+	var currentname = elem.value || "";
+        var usernameblock = new OO.ui.infuse(elem.parentElement.parentElement.parentElement.parentElement);
+	// Start with username input field, and go up 4 levels, to the entire username container that OOUI will infuse onto
+	var noticebox = [];
+	if(currentname.length > 0 && currentname[0].match("[a-z]")){// Compare first letter to a regex, to check if it starts with a lowercase letter
+		noticebox[0] = new mw.message("createacct-normalization", "", currentname[0].toUpperCase() + currentname.slice(1)).text();
```

Comment:
Ideally we would make some username-related API call and defer the normalization to MediaWiki instead of hardcoding this behavior in JS.

Reply:
Do you have a particular call in mind? As far as I know the standard ones wouldn't work because it needs to handle the Scratch limits as well. There's also optimization to consider: whether it'd be worth it to make a remote call which would make it show up slower and add more calls when it could show up instantly and this is unlikely to ever change. In my opinion it can be handled locally and it's just "hardcoding" the same thing a remote server would do. But if there are good reasons to remote it, then maybe?

Reply:
This wasn't a change request, my intent was more to note this for future reference, which I realize I didn't make clear.